src/api.js

```diff
@@ -229,6 +238,74 @@ class DICOMwebClient {
         }
       }
 
+      // Event triggered while upload progresses
+      if ('uploadCallbacks' in options) {
+        const {uploadCallbacks} = options;
```

Comment:
I would refactor this into a private method inside the class as it is adding a lot of noise for someone who doesn't have that callbacks

```js
if ('uploadCallbacks' in options) {
    this._registerUploadCallbacks(...)
}
```